novawallet/Modules/NetworkDetails/NetworkDetailsInteractor.swift

```diff
@@ -0,0 +1,176 @@
+import UIKit
+import SubstrateSdk
+import SoraFoundation
+import Operation_iOS
+
+final class NetworkDetailsInteractor {
+    weak var presenter: NetworkDetailsInteractorOutputProtocol?
+
+    private var chain: ChainModel
+    private let connectionFactory: ConnectionFactoryProtocol
+    private let chainRegistry: ChainRegistryProtocol
+    private let repository: AnyDataProviderRepository<ChainModel>
+    private let operationQueue: OperationQueue
+    private let nodeMeasureQueue: OperationQueue
+
+    private var nodesConnections: [String: ChainConnection] = [:]
+
+    init(
+        chain: ChainModel,
+        connectionFactory: ConnectionFactoryProtocol,
+        chainRegistry: ChainRegistryProtocol,
+        repository: AnyDataProviderRepository<ChainModel>,
+        operationQueue: OperationQueue,
+        nodeMeasureQueue: OperationQueue
+    ) {
+        self.chain = chain
+        self.connectionFactory = connectionFactory
+        self.chainRegistry = chainRegistry
+        self.repository = repository
+        self.operationQueue = operationQueue
+        self.nodeMeasureQueue = nodeMeasureQueue
+    }
+}
+
+// MARK: NetworkDetailsInteractorInputProtocol
+
+extension NetworkDetailsInteractor: NetworkDetailsInteractorInputProtocol {
+    func setup() {
+        connectToChainNodes()
+        subscribeChainChanges()
+    }
+
+    func toggleNetwork() {
```

Comment:
I wouldn't use toggle logic here since we executing it asynchronously. Let's instead pass a concrete value here.